### PR TITLE
better TLS1.3 check

### LINKED ISSUE #10285: [Bug] 1.3 Cannot start farmer on a Mac with python3 from Xcode
### What happened?

Latest version 1.2.12.dev309 on a Mac running BigSur 11.5.2, and latest version of Xcode (13.2.1).

Here is the error when running chia start farmer

```python-traceback
 File "/Users/stipus/chia-blockchain/chia/daemon/server.py", line 1478, in async_run_daemon
    await ws_server.start()
  File "/Users/stipus/chia-blockchain/chia/daemon/server.py", line 171, in start
    self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_3
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/ssl.py", line 586, in minimum_version
    super(SSLContext, SSLContext).minimum_version.__set__(self, value)
ValueError: Unsupported protocol version 0x304
Daemon not started yet
Failed to create the chia daemon
```

I had to edit server.py line 171 to 
```python
 self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_2
```

### Version

1.2.12.dev309

### What platform are you using?

macOS

### What ui mode are you using?

CLI

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by emlowe**
A better and more compatible TLS check that works on versions of python using a non-openssl TLS library. Examples include the python3 version in Xcode which is using LibreSSL.

See issue #10285

### CODE REVIEW

**Review by hoffmang9 — DISMISSED** [review]
lgtm

**Comment by emlowe** [comment]
Working on some minor code cleanups

File: chia/daemon/server.py [hunk 1]
```diff
@@ -156,18 +156,28 @@ def __init__(
...
             if self.ssl_context is not None:
-                # Daemon is internal connections, so override to TLS1.3 only
-                self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_3
+                try:
+                    # Daemon is internal connections, so override to TLS1.3 only
+                    self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_3
+                except ValueError:
+                    self.log.warning(
+                        (
+                            "Deprecation Warning: Your version of SSL (%s) does not support TLS1.3. "
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/daemon/server.py]: I have no idea what I did, but it made black and mypy happy, so I was happy

File: chia/daemon/server.py [hunk 2]
```diff
@@ -156,18 +156,28 @@ def __init__(
...
+                            "A future version of Chia will require TLS1.3."
+                        ),
+                        ssl.OPENSSL_VERSION,
+                    )
+                    self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_2
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/daemon/server.py]: Would have already thrown earlier when this is set by a function called during init

File: chia/daemon/server.py [hunk 3]
```diff
@@ -156,18 +156,28 @@ def __init__(
     async def start(self):
         self.log.info("Starting Daemon Server")
 
-        if ssl.OPENSSL_VERSION_NUMBER < 0x10101000:
+        if not ssl.HAS_TLSv1_3:
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/daemon/server.py]: I like it, yes. personal style choice probably.

File: chia/daemon/server.py [hunk 4]
```diff
@@ -156,18 +156,23 @@ def __init__(
...
 
-        if ssl.OPENSSL_VERSION_NUMBER < 0x10101000:
+        # Note: the minimum_version has been already set to TLSv1_2
+        # in ssl_context_for_server()
+        # Daemon is internal connections, so override to TLS1.3 only
+        if ssl.HAS_TLSv1_3:
+            try:
+                self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_3
+            except ValueError:
+                self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_2  # probably not needed but doesn't hurt
```
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/daemon/server.py]: I made a small change here for comment nits

File: chia/daemon/server.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (emlowe) [reply chia/daemon/server.py]: And also, I'm old C programmer to printf style FTW!

**Review by altendky — APPROVED** [review]

**Pull request merged**